File: problems/2599_make-the-prefix-sum-non-negative/solution.py
from typing import List, Optional

import heapq


class Solution:
    def makePrefSumNonNegative(self, nums: List[int]) -> int:
        heap = []
        prefix = 0
        ops = 0
        for num in nums:
            prefix += num
            heapq.heappush(heap, num)
            while prefix < 0:
                prefix -= heapq.heappop(heap)
                ops += 1
        return ops

File: problems/2599_make-the-prefix-sum-non-negative/solution.java
import java.util.PriorityQueue;

class Solution {

    public int makePrefSumNonNegative(int[] nums) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        long prefix = 0;
        int ops = 0;
        for (int num : nums) {
            prefix += num;
            heap.offer(num);
            while (prefix < 0) {
                prefix -= heap.poll();
                ops++;
            }
        }
        return ops;
    }
}
